masq/src/commands/setup_command.rs

```diff
@@ -372,38 +351,103 @@ ip                            No sir, I don't like it.\n\
+                note_expected: false,
+                status_chain: UiSetupResponseValueStatus::Default,
+                status_data_dir: UiSetupResponseValueStatus::Default,
+            },
+            SetupCommandData {
+                chain_str: Some("polygon-mumbai".to_owned()),
+                data_directory: None,
+                chain_name_expected: Some("polygon-mumbai"),
+                data_directory_expected: Some("/home/cooga/.local/MASQ/polygon-mumbai"),
+                note_expected: true,
+                status_chain: UiSetupResponseValueStatus::Set,
+                status_data_dir: UiSetupResponseValueStatus::Default,
+            },
+            SetupCommandData {
+                chain_str: None,
+                data_directory: Some("booga".to_owned()),
+                chain_name_expected: Some("polygon-mainnet"),
+                data_directory_expected: Some("booga/polygon-mainnet"),
+                note_expected: true,
+                status_chain: UiSetupResponseValueStatus::Default,
+                status_data_dir: UiSetupResponseValueStatus::Set,
+            },
+            SetupCommandData {
+                chain_str: Some("polygon-mumbai".to_owned()),
+                data_directory: Some("booga/polygon-mumbai".to_owned()),
+                chain_name_expected: Some("polygon-mumbai"),
+                data_directory_expected: Some("booga/polygon-mumbai/polygon-mumbai"),
+                note_expected: true,
+                status_chain: UiSetupResponseValueStatus::Set,
+                status_data_dir: UiSetupResponseValueStatus::Set,
+            },
         ].iter().for_each(|
-            (chain_opt,
-             data_directory_opt,
-             chain_name_expected,
-             data_directory_expected,
-             note_expected,
-             status_chain,
-             status_data_dir)| {
-            let note_expected_real = match &*note_expected {
-                Some(..) => { match note_expected.unwrap() { Some(..) => note_expected.unwrap().unwrap(), None => "" } },
+            data| {
+            let note_expected_real = match data.note_expected {
+                true => "\nNOTE: your data directory was modified to match the chain parameter.\n",
                 _ => ""
             };
-            let status_data_dir_str = match *status_data_dir { Default => "Default", Set => "Set", Configured => "Configured", UiSetupResponseValueStatus::Blank => "Blank", UiSetupResponseValueStatus::Required => "Required" };
-            let status_chain_str = match *status_chain { Default => "Default", Set => "Set", Configured => "Configured", UiSetupResponseValueStatus::Blank => "Blank", UiSetupResponseValueStatus::Required => "Required" };
+            let status_data_dir_str = match data.status_data_dir {
+                Default => "Default",
+                Set => "Set",
+                Configured => "Configured",
+                UiSetupResponseValueStatus::Blank => "Blank",
```

Comment:
This isn't wrong, but why do you need the `Blank` and `Required` values to be qualified with `UiSetupResponseValueStatus::` but not the others? Curious.